From fb1debe5f6a74f76a68594d7952367005986d025 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 18 Jan 2024 16:49:46 +0100
Subject: [PATCH 0494/1695] plat-stm32mp1: scmi server may not embed voltage
 domain service

SCMI voltage domain protocol is not exposed when
CFG_SCMI_MSG_VOLTAGE_DOMAIN is disabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e9fd617d8c5957ee73816efbaebd87d8856a3e48 plat-stm32mp1: scmi server may not embed voltage domain service]
Change-Id: Ic2710b7048dc5f7dbf44639eb4a71ae910613a63
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/352528
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/scmi_server.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/scmi_server.c b/core/arch/arm/plat-stm32mp1/scmi_server.c
index 3f7b1ca57..5a525125e 100644
--- a/core/arch/arm/plat-stm32mp1/scmi_server.c
+++ b/core/arch/arm/plat-stm32mp1/scmi_server.c
@@ -221,7 +221,7 @@ static struct stm32_scmi_rd stm32_scmi_reset_domain[] = {
 };
 #endif
 
-#ifndef CFG_SCMI_MSG_REGULATOR_CONSUMER
+#if defined(CFG_SCMI_MSG_VOLTAGE_DOMAIN) && !defined(CFG_SCMI_MSG_REGULATOR_CONSUMER)
 #ifdef CFG_STM32MP13
 struct stm32_scmi_voltd scmi_voltage_domain[] = {
 	VOLTD_CELL_PWR(VOLTD_SCMI_REG11, PWR_REG11, "reg11"),
@@ -268,7 +268,7 @@ struct stm32_scmi_voltd scmi_voltage_domain[] = {
 	VOLTD_CELL_PMIC(VOLTD_SCMI_STPMIC1_PWR_SW2, "pwr_sw2", "vbus_sw"),
 };
 #endif
-#endif /*!CFG_SCMI_MSG_REGULATOR_CONSUMER*/
+#endif /*CFG_SCMI_MSG_VOLTAGE_DOMAIN && !CFG_SCMI_MSG_REGULATOR_CONSUMER*/
 
 struct channel_resources {
 	struct scmi_msg_channel *channel;
@@ -292,7 +292,7 @@ static const struct channel_resources scmi_channel[] = {
 		.clock_count = ARRAY_SIZE(stm32_scmi_clock),
 		.rd = stm32_scmi_reset_domain,
 		.rd_count = ARRAY_SIZE(stm32_scmi_reset_domain),
-#ifndef CFG_SCMI_MSG_REGULATOR_CONSUMER
+#if defined(CFG_SCMI_MSG_VOLTAGE_DOMAIN) && !defined(CFG_SCMI_MSG_REGULATOR_CONSUMER)
 		.voltd = scmi_voltage_domain,
 		.voltd_count = ARRAY_SIZE(scmi_voltage_domain),
 #endif
@@ -366,7 +366,9 @@ const char *plat_scmi_sub_vendor_name(void)
 static const uint8_t plat_protocol_list[] = {
 	SCMI_PROTOCOL_ID_CLOCK,
 	SCMI_PROTOCOL_ID_RESET_DOMAIN,
+#ifdef CFG_SCMI_MSG_VOLTAGE_DOMAIN
 	SCMI_PROTOCOL_ID_VOLTAGE_DOMAIN,
+#endif
 	0 /* Null termination */
 };
 
-- 
2.47.2

